crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -3551,6 +3667,26 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_read_only_tx_shared_lock() -> ResultTest<()> {
```

Comment:
I was about to leave a comment asking for tests, but you beat me to it! I feel good about this test.
```suggestion
    /// Test that two read-only TXes can operate concurrently without deadlock or blocking,
    /// and that both observe correct results for a simple table scan.
    fn test_read_only_tx_shared_lock() -> ResultTest<()> {
```